src/meta/service/src/network.rs

```diff
@@ -186,21 +276,38 @@ impl RaftNetwork<LogEntry> for Network {
         info!("send_vote: target: {} rpc: {}", target, rpc.summary());
 
         let mut client = self.make_client(&target).await?;
-        let req = common_tracing::inject_span_to_tonic_request(rpc);
-
-        self.incr_meta_metrics_sent_bytes_to_peer(&target, req.get_ref());
 
-        let resp = client.vote(req).await;
-        info!("vote: resp from target={} {:?}", target, resp);
+        let mut mes = Default::default();
+        let mut last_err = None;
+        for back_off in self.back_off() {
+            let req = common_tracing::inject_span_to_tonic_request(&rpc);
+
+            self.incr_meta_metrics_sent_bytes_to_peer(&target, req.get_ref());
+
+            let resp = client.vote(req).await;
+            info!("vote: resp from target={} {:?}", target, resp);
+
+            match resp {
+                Ok(resp) => {
+                    mes = resp.into_inner();
+                    let resp = serde_json::from_str(&mes.data)?;
+                    return Ok(resp);
+                }
+                Err(e) => {
+                    incr_meta_metrics_sent_failure_to_peer(&target);
+                    last_err = Some(e);
+                    // back off and retry
+                    sleep(back_off).await;
+                }
+            }
+        }
 
-        if resp.is_err() {
-            incr_meta_metrics_sent_failure_to_peer(&target);
+        if let Some(e) = last_err {
+            // all back-offed requests failed
+            return Err(anyhow::Error::from(e));
         }
 
-        let resp = resp?;
-        let mes = resp.into_inner();
         let resp = serde_json::from_str(&mes.data)?;
-
         Ok(resp)
```

Comment:
Why is there a chance the loop of backoff finished, but there isn't an error?

Reply:
If you're asking why there is always at least one chance being consumed, that's because we are using the `ExponentialBackoff` structure directly and for each iteration, a chance will be consumed. So this is unavoidable since at least a loop should run to get a response.

If you are suggesting that the parsing error should also trigger a backoff, the newly pushed commits should fill your satisfaction.